pkg/prometheus/operator.go

```diff
@@ -247,3 +278,36 @@ func (sr *StatusReporter) Process(ctx context.Context, p monitoringv1.Prometheus
 
 	return &pStatus, nil
 }
+
+// UpdateServiceMonitorStatus add the latest status in serviceMonitor selected by the Prometheus or PrometheusAgent.
+func UpdateServiceMonitorStatus(ctx context.Context, p metav1.Object, c *ConfigResourceSyncer, res TypedConfigurationResource[*monitoringv1.ServiceMonitor]) error {
```

Comment:
(nit) I would put the resource syncer as the second argument
```suggestion
func UpdateServiceMonitorStatus(
    ctx context.Context,
    c *ConfigResourceSyncer,
    p metav1.Object,
    res TypedConfigurationResource[*monitoringv1.ServiceMonitor]) error {
```